### PR TITLE
feat: add setters to AllowlistMarketHook

### PR DESCRIPTION
**Pull request by ramenforbreakfast**

### CODE REVIEW

**Review by davidprepo — APPROVED** [review]
LGTM

File: apps/smart-contracts/core/test/AllowlistMarketHook.test.ts [hunk 1]
```diff
@@ -40,4 +39,112 @@ describe('=> AllowlistMarketHook', () => {
...
+    it('blocks a market', async () => {
+      await allowlistMarketHook.connect(deployer).setAllowedMarket(deployer.address, true)
+      expect(await allowlistMarketHook.isMarketAllowed(deployer.address)).to.eq(true)
+
+      await allowlistMarketHook.connect(deployer).setAllowedMarket(deployer.address, false)
+
+      expect(await allowlistMarketHook.isMarketAllowed(deployer.address)).to.eq(false)
+    })
+
+    it('is idempotent when allowing', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/AllowlistMarketHook.test.ts:73]: nit: if not when
same for below

**Review by XavierEkkel — APPROVED** [review]

**Comment by ramenforbreakfast** [comment]
I will re-request a review once PR #539 is modified according to the technical spec and most likely introduce a batch setter.

**Comment by ramenforbreakfast** [comment]
Closing since implementation has changed significantly, can likely reuse some of this code though.

**Pull request closed without merging**